### PR TITLE
GithubCI: use Ubuntu 24.04 instead of 23.10

### PR DESCRIPTION
**Pull request by webwarrior-ws**
Because 23.10 is no longer supported and 24.04 is LTS release.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -32,6 +31,10 @@ jobs:
           # The remote certificate is invalid because of errors in the certificate chain: UntrustedRoot
           apt install --yes --no-install-recommends ca-certificates
 
+          # register backports repository (needed for dotnet6)
+          sudo apt install  --yes --no-install-recommends software-properties-common
+          sudo add-apt-repository ppa:dotnet/backports
```
Comment (knocte) [inline .github/workflows/CI.yml:36]: @webwarrior-ws this is unacceptable, we don't want to use PPAs

Comment (webwarrior-ws) [reply .github/workflows/CI.yml:36]: How are we going to install .NET 6 then?

Comment (knocte) [reply .github/workflows/CI.yml:36]: well, if dotnet6 is not available in 24.04 then we upgrade

Comment (webwarrior-ws) [reply .github/workflows/CI.yml]: Upgraded to .NET 8

**Pull request merged**